ada/ast.py

```diff
@@ -12881,20 +12949,41 @@ class MultiDimArrayAssoc(AggregateAssoc):
     pass
 
 
-class ParamActual(Struct):
-    """
-    Data structure used by zip_with_params property. Associates an expression
-    (the actual) to a formal param declaration (the parameter).
-    """
-    param = UserField(type=T.DefiningName.entity)
-    actual = UserField(type=T.Expr.entity)
-
-
 class AssocList(BasicAssoc.list):
     """
     List of associations.
     """
 
+    @langkit_property(return_type=T.Expr.entity)
+    def actual_for_param_at(param=T.DefiningName.entity,
+                            pos=T.Int,
+                            default_expr=(T.Expr.entity, No(T.Expr.entity))):
+        """
+        Return the actual expression for ``param`` if any, ``default_expr``
+        otherwise.
+        """
+        up = Var(Entity.unpacked_params)
```

Comment:
It feels very sub-optimal to rebuild the whole flattened list of `unpacked_params` each time this property is called, which means each time we treat one argument in the calling property. Could there be a way to avoid it? Maybe by passing it as parameter, or even memoizing it.

Reply:
I chose to memoize it!